src/upgrader.js

```diff
@@ -268,8 +269,13 @@ class Upgrader {
     maConn.timeline = new Proxy(_timeline, {
       set: (...args) => {
         if (connection && args[1] === 'close' && args[2] && !_timeline.close) {
-          connection.stat.status = 'closed'
-          this.onConnectionEnd(connection)
+          // Wait for close to finish before notifying of the closure
+          (async () => {
```

Comment:
Can this anonymous function throw?  The error needs to be handled if so.

Reply:
Yes, in theory the connection.close could throw although that should be rare. Good catch, a handler for this to log the error